### PR TITLE
chore: Cleanup and reorganize interfaces for components

### PR DESCRIPTION
**Pull request by driskull**
## Summary

Cleanup and reorganize interfaces for components

- Removes 'interfaces' folder in 'src' in favor of `interfaces` files inside component folders
- There were two `common` interface files, now there is just one `interfaces` file inside `/components` folder.
- uses types in component files from common interface files
- Removes use of 'Calcite' in type names.

### CODE REVIEW

File: src/components/calcite-inline-editable/calcite-inline-editable.e2e.ts [hunk 1]
```diff
@@ -253,7 +253,7 @@ describe("calcite-inline-editable", () => {
 
     it("disables editing when afterConfirm resolves successfully", async () => {
       const element = await page.find("calcite-inline-editable");
-      const afterConfirm = () => new Promise((resolve) => setTimeout(resolve, 100));
+      const afterConfirm: () => Promise<void> = () => new Promise((resolve) => setTimeout(resolve, 100));
```
Comment (jcfranco) [inline src/components/calcite-inline-editable/calcite-inline-editable.e2e.ts:256]: Nice 👍

File: src/components/calcite-dropdown-item/calcite-dropdown-item.tsx [hunk 2]
```diff
@@ -37,7 +38,7 @@ export class CalciteDropdownItem {
   @Prop({ reflect: true, mutable: true }) active = false;
 
   /** flip the icon(s) in rtl */
-  @Prop({ reflect: true }) iconFlipRtl?: "both" | "start" | "end";
+  @Prop({ reflect: true }) iconFlipRtl?: Flip;
```
Comment (jcfranco) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.tsx:41]: Nitpick: `RtlFlip`, `FlipContext` or similar?

**Review by jcfranco — APPROVED** [review]
Awesome cleanup! Couple of notes:

* Regarding interface naming: when should it be prefixed by the component name? For example, `<Component>Alignment` vs `Alignment`. I like the concise option, personally, but I'm flexible. 😄 
* Prop values from READMEs will be obscured from this as the doc won't inline the type values. E.g., you'll get `Scale` instead of `"s" | "m" | "l"`. Something to keep in mind since there's not much we can do here (Stencil docgen limitation).

**Review by paulcpederson — APPROVED** [review]

**Comment by paulcpederson** [comment]
🔥 🔥 🔥

File: src/components/calcite-dropdown-item/calcite-dropdown-item.tsx (hunk 2 above)
Comment (driskull) [reply src/components/calcite-dropdown-item/calcite-dropdown-item.tsx:41]: FlipContext 👍

File: src/components/calcite-inline-editable/calcite-inline-editable.e2e.ts (hunk 1 above)
Comment (driskull) [reply src/components/calcite-inline-editable/calcite-inline-editable.e2e.ts:256]: yeah this kept popping up randomly. needed a type

**Comment by driskull** [comment]
> Regarding interface naming: when should it be prefixed by the component name?

If it is used by multiple components then it can be in the root interfaces file and have no prefix. If it only applies to a single set of components then it should be in the local interfaces and prefixed. Sound right?

> Prop values from READMEs will be obscured from this as the doc won't inline the type values. E.g., you'll get Scale instead of "s" | "m" | "l". Something to keep in mind since there's not much we can do here (Stencil docgen limitation)

Yeah. :(

**Comment by jcfranco** [comment]
> If it is used by multiple components then it can be in the root interfaces file and have no prefix. If it only applies to a single set of components then it should be in the local interfaces and prefixed. Sound right?

👍

**Pull request merged**